src/NodeApi/Runtime/NativeLibrary.cs

```diff
@@ -68,25 +115,77 @@ public static nint GetExport(nint handle, string name)
     public static bool TryGetExport(nint handle, string name, out nint procAddress)
     {
 #if NETFRAMEWORK || NETSTANDARD
-        procAddress = GetProcAddress(handle, name);
-        return procAddress != default;
+        procAddress = GetSymbol(handle, name, throwOnError: false);
+        return procAddress != 0;
 #else
         return SysNativeLibrary.TryGetExport(handle, name, out procAddress);
 #endif
     }
 
+    static nint GetSymbol(nint handle, string name, bool throwOnError)
+    {
+        if (handle == 0)
+            throw new ArgumentNullException(nameof(handle));
+        if (string.IsNullOrEmpty(name))
+            throw new ArgumentNullException(nameof(name));
+
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            nint procAddress = GetProcAddress(handle, name);
+            if (procAddress == 0 && throwOnError)
+                throw new DllNotFoundException(new Win32Exception(Marshal.GetLastWin32Error()).Message);
+
+            return procAddress;
+        }
+        else
+        {
+            dlerror();
+            nint procAddress = dlsym(handle, name);
+            nint error = dlerror();
+            if (error != 0)
+            {
+                if (throwOnError)
+                    throw new EntryPointNotFoundException(Marshal.PtrToStringAuto(error));
+
+                procAddress = 0;
+            }
+
+            return procAddress;
+        }
+    }
+
 #pragma warning disable CA2101 // Specify marshaling for P/Invoke string arguments
 
     [DllImport("kernel32")]
     private static extern nint GetModuleHandle(string? moduleName);
 
-    [DllImport("kernel32")]
+    [DllImport("kernel32", SetLastError = true)]
     private static extern nint LoadLibrary(string moduleName);
 
-    [DllImport("kernel32")]
+    [DllImport("kernel32", SetLastError = true)]
     private static extern nint GetProcAddress(nint hModule, string procName);
 
-    private static nint dlopen(nint fileName, int flags)
+    private static nint dlerror()
+    {
+        // Some Linux distros / versions have libdl version 2 only.
+        // Mac OS only has the unversioned library.
+        try
+        {
+            return dlerror2();
+        }
+        catch (DllNotFoundException)
+        {
+            return dlerror1();
+        }
+    }
+
+    [DllImport("libdl", EntryPoint = "dlerror")]
+    private static extern nint dlerror1();
+
+    [DllImport("libdl.so.2", EntryPoint = "dlerror")]
+    private static extern nint dlerror2();
+
+    private static nint dlopen(string fileName, int flags)
```

Comment:
I am getting a warning on the line 35 above where we pass `default` as the `fileName`. Making the first parameter nullable fixes the issue: `string? fileName`.

Reply:
True. Fixed.